thriftbp/tracing.go

```diff
@@ -0,0 +1,83 @@
+package thriftbp
+
+import (
+	"context"
+	"strconv"
+
+	"github.com/apache/thrift/lib/go/thrift"
+
+	"github.com/reddit/baseplate.go/set"
+	"github.com/reddit/baseplate.go/tracing"
+)
+
+// CreateThriftContextFromSpan injects span info into a context object that can
+// be used in thrift client code.  If you are using a client pool created using
+// thriftclient.NewBaseplateClientPool, all of your thrift calls will already be
+// call this automatically, so there is no need to use it directly.
+//
+// Caller should first create a client child-span for the thrift call as usual,
+// then use that span and the parent context object with this call,
+// then use the returned context object in the thrift call.
+// Something like:
+//
+//     span, clientCtx := opentracing.StartSpanFromContext(
+//       ctx,
+//       "myCall",
+//       tracing.SpanTypeOption{Type: tracing.SpanTypeClient},
+//     )
+//     result, err := client.MyCall(clientCtx, arg1, arg2)
```

Comment:
There's one step missing before this line:

```
clientCtx = thriftbp.CreateThriftContextFromSpan(clientCtx, traceing.AsSpan(span))
```

Or maybe inline that into the client call.